GGRC-5955 Activate Workflow button is enabled if create  Workflow with empty Task Group

MikalaiMikalalai opened the pull request:
<!-- If your PR depends on other tickets or PRs, you can list them here
# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] GGRC-1234
- [ ] #1234
-->

# Issue description

Activate Workflow button is enabled if create  Workflow with empty Task Group

# Steps to test the changes

1. Create Repeat off workflow with empty FIRST TASK GROUP'S TITLE field
2. Open the workflow
3. Look at the 'Activate Workflow' button

Actual Result: Activate Workflow button is enabled if create Workflow with empty Task Group
Expected Result: Activate Workflow button should not be enabled if create Workflow with empty Task Group

# Solution description

Add check for empty task_groups

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [x] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [x] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
- [ ] Upon merging, add 'check migration chain' and 'kokoro:force-run' labels to all open PRs with a label 'migration'
- [ ] Upon merging, update 'table structure changes' in weekly deployment documentation
-->

# PR Review checklist

- [x] The changes fix the issue and don't cause any apparent regressions.
- [x] Labels and milestone are correctly set.
- [x] The solution description matches the changes in the code.
- [x] There is no apparent way to improve the performance & design of the new code.
- [x] The pull request is opened against the correct base branch.
- [x] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

mashabarashko commented:
@MikalaiMikalalai I still have 500 error locally when hit Activate button:
<img width="621" alt="Screen Shot 2019-05-13 at 12 09 59 PM" src="https://user-images.githubusercontent.com/2690200/57609712-16c44e80-7578-11e9-9834-17c94487d02c.png">
Could you please investigate?

mashabarashko commented:
@MikalaiMikalalai Button is disabled, however I suppose this 500 error will be returned if we send api query. I think, it should be fixed.

MikalaiMikalalai commented:
> @MikalaiMikalalai Button is disabled, however I suppose this 500 error will be returned if we send api query. I think, it should be fixed.

You are right, I fixed this issue and updated test. Thanks

mashabarashko commented on src/ggrc_workflows/__init__.py:
```diff
@@ -196,6 +196,9 @@ def handle_cycle_post(sender, obj=None, src=None, service=None):  # noqa pylint:
   workflow.status = workflow.ACTIVE
+  if not workflow.can_start_cycle:
+    raise ValueError("Workflow with misconfigured "
+                     "Task Groups can not be activated.")
```
@MikalaiMikalalai I think slightly better to check it before line 198

mashabarashko replied:
@MikalaiMikalalai I think slightly better to check it before line 198
